### PR TITLE
[CalloutCard] add Iconable actions & allows Button variant for secondary action

### PR DESCRIPTION
**Pull request by arthurgouveia**
Trying to make sure we can get this design going, like we already have for [Search and Discovery](https://github.com/Shopify/discovery-app/blob/main/app/ui/components/FeedbackCard/FeedbackCard.tsx) without needing to build custom components.

We'll need that now on Email too and if this is approved, we could just have standard code everywhere for collecting feedback.

Screenshot 2024-01-30 at 2 25 36 PM

### WHY are these changes introduced and WHAT is this pull request doing?

Currently we don't allow `primaryAction` to be null, which would allow us to achieve the same results by using `content`. I figured we could just enhance the API slightly making it backwards compatible and allowing more customization.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [x] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: .changeset/popular-birds-refuse.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+[CalloutCard] Adds IconableActions & 2ndary actions can have variants
```
Comment (kyledurand) [inline .changeset/popular-birds-refuse.md]:
```suggestion
[CalloutCard] Added IconableAction to primaryAction and secondaryAction
[CalloutCard] Added variant prop to secondaryAction
```

**Review by kyledurand — APPROVED** [review]

File: polaris-react/src/components/CalloutCard/CalloutCard.tsx [hunk 2]
```diff
@@ -40,7 +40,7 @@ export function CalloutCard({
   const primaryActionMarkup = buttonFrom(primaryAction);
   const secondaryActionMarkup = secondaryAction
     ? buttonFrom(secondaryAction, {
-        variant: 'tertiary',
+        variant: secondaryAction.variant || 'tertiary',
```
Comment (kyledurand) [inline polaris-react/src/components/CalloutCard/CalloutCard.tsx]: Shouldn't matter in this case but we use nullish coalescing for this type of thing

```suggestion
        variant: secondaryAction.variant ?? 'tertiary',
```

**Comment by arthurgouveia** [comment]
Screenshot 2024-02-05 at 4 04 07 PM
Screenshot 2024-02-05 at 4 04 19 PM

Screenshot 2024-02-05 at 4 10 47 PM

**Pull request merged**